lib/client-assets.php

```diff
@@ -686,84 +686,84 @@ function gutenberg_default_script_modules() {
 			'fetchpriority' => 'low',
 		);
 
-		gutenberg_register_interactive_script_module_id( $script_module_id );
-
 		$path = gutenberg_url( "build-module/{$file_name}" );
 		wp_register_script_module( $script_module_id, $path, $script_module_data['dependencies'], $script_module_data['version'], $args ); // The $args parameter is new as of WP 6.9 per <https://core.trac.wordpress.org/ticket/61734>.
 	}
 }
 remove_action( 'wp_default_scripts', 'wp_default_script_modules' );
 add_action( 'wp_default_scripts', 'gutenberg_default_script_modules' );
 
-/*
- * Always remove the Core action hook while gutenberg_enqueue_stored_styles() exists to avoid styles being printed twice.
- * This is also because gutenberg_enqueue_stored_styles uses the Style Engine's `gutenberg_*` functions and `_Gutenberg` classes,
- * which are in continuous development and generally ahead of Core.
- */
-remove_action( 'wp_enqueue_scripts', 'wp_enqueue_stored_styles' );
-remove_action( 'wp_footer', 'wp_enqueue_stored_styles', 1 );
+if ( version_compare( get_bloginfo( 'version' ), '6.9.0', '<' ) ) {
+	/**
+	 * Access the shared static variable for interactive script modules.
+	 *
+	 * @param string|null $script_module_id The script module ID to register, or null to get the list.
+	 * @return array<string, true> Associative array of script module ID => true.
```

Comment:
Might want to make this abundantly clear that this "script module ID" is the ID for the `SCRIPT[type=module]` _tag_, not the registered "Script Module". Both are being used in this function, which may be confusing.